base/applications/notepad/text.c

```diff
@@ -98,6 +225,10 @@ ReadText(HANDLE hFile, LPWSTR *ppszText, DWORD *pdwTextLen, ENCODING *pencFile,
         encFile = ENCODING_UTF8;
         dwPos += 3;
     }
+    else
+    {
+        encFile = AnalyzeEncoding(pBytes, dwSize);
```

Comment:
This (and all the stuff above) seems out of place here. Shouldn't notepad just call `IsTextUnicode`?

Reply:
`IsTextUnicode` can be used for UTF-16 text. And cannot be used for UTF-8 text.

Reply:
Ah. Fair.

Reply:
I used `MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, ...)`.